clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocol.java

```diff
@@ -224,14 +230,14 @@ public static ByteBuffer serializeSubscription(PartitionAssignor.Subscription su
 
         switch (version) {
             case CONSUMER_PROTOCOL_V0:
-                return buildSubscriptionV0(buffer, groupInstanceId);
+                return deserializeSubscriptionV0(buffer);
 
             case CONSUMER_PROTOCOL_V1:
-                return buildSubscriptionV1(buffer, groupInstanceId);
+                return deserializeSubscriptionV1(buffer);
 
             // assume all higher versions can be parsed as V1
             default:
-                return buildSubscriptionV1(buffer, groupInstanceId);
+                return deserializeSubscriptionV1(buffer);
```

Comment:
nit: Although not strictly part of the PR, this part isn't covered in the unit test. Is it worth mocking out future version /different protocol to make sure this works as intended?  Same for `serializeSubscription`

Reply:
Could you elaborate a bit more? I thought future-proof is covered in deserializeFutureSubscription/AssignmentVersion`.

Reply:
My bad,  I missed the tests `deserializeFutureSubscription/AssignmentVersion` and was looking at the coverage of `de/serializeAssignment`, and the `default` branch isn't hit, but the individual methods themselves are covered.